analysis.ipynb

```diff
@@ -0,0 +1,1563 @@
+{
```

Comment:
why not use column_transformer instead of custom made def preprocess_inputs(df)?

Reply:
ColumnTransformer is used when loading data into the model. But what is done here is preprocessing, or better called feature selection and feature engineering.

Reply:
yea, sure that's fine too. But the grouping code grouped all category to other, can you look into that?

Reply:
Yeah, that was intentional, because we want to avoid having hundreds of columns with 99% 0 values.

Reply:
I understand what you trying to do. but there may be problem that the result of group is 1 category which has no predicting power.